// rtlink-mbridge/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  checkoutNetwork: () => checkoutNetwork3,
  default: () => index_default,
  getDeviceInfo: () => getDeviceInfo3,
  isAndroid: () => isAndroid,
  isIOS: () => isIOS,
  preDial: () => preDial3,
  readLocal: () => readLocal3,
  scan: () => scan3,
  setScreenHorizontal: () => setScreenHorizontal3,
  setScreenPortrait: () => setScreenPortrait3,
  writeLocal: () => writeLocal3
});
module.exports = __toCommonJS(index_exports);

// package.json
var package_default = {
  name: "rtlink-mbridge",
  version: "0.26.0",
  description: "",
  main: "index.js",
  module: "./dist/index.mjs",
  types: "./dist/index.d.ts",
  exports: {
    ".": {
      import: "./dist/index.mjs",
      require: "./dist/index.js",
      types: "./dist/index.d.ts"
    }
  },
  files: ["dist"],
  scripts: {
    test: 'echo "Error: no test specified" && exit 1',
    build: "tsup src/index.ts --format cjs,esm --dts"
  },
  keywords: [],
  author: "",
  license: "ISC",
  devDependencies: {
    tsup: "^8.5.1",
    typescript: "^5.9.3"
  }
};

// src/utils.ts
var fnSpliter = "|rtm|";
function isIOS() {
  const ua = navigator.userAgent.toLowerCase();
  if (/iphone|ipad|ipod/.test(ua)) {
    return true;
  }
  if (navigator.platform === "MacIntel" && navigator.maxTouchPoints > 1) {
    return true;
  }
  if (navigator.platform.includes("iPhone") || navigator.platform.includes("iPad")) {
    return true;
  }
  return false;
}
function isAndroid() {
  if (typeof navigator === "undefined") return false;
  return /android/i.test(navigator.userAgent);
}

// src/iOS.ts
var postDataToiOS = (data) => {
  const dataType = typeof data;
  let _data;
  if (dataType == "object") {
    _data = JSON.stringify(data);
  } else {
    _data = data;
  }
  try {
    window.webkit?.messageHandlers?.swiftHandler.postMessage(_data);
  } catch (err) {
    console.error(err);
  }
};
var getDeviceInfo = async () => {
  return new Promise((resolve) => {
    postDataToiOS(`getDeviceInfo`);
    window.rtmobile.callbacks.afterGetDeviceInfo = resolve;
  });
};
var writeLocal = async (key, value, seconds) => {
  return new Promise((resolve) => {
    postDataToiOS(`writeLocal${fnSpliter}${key}${fnSpliter}${value}${fnSpliter}${seconds}`);
    window.rtmobile.callbacks.afterWriteLocal = resolve;
  });
};
var readLocal = async (key) => {
  return new Promise((resolve) => {
    postDataToiOS(`readLocal${fnSpliter}${key}`);
    window.rtmobile.callbacks.afterReadLocal = resolve;
  });
};
var preDial = async (number) => {
  postDataToiOS(`preDial${fnSpliter}${number}`);
};
var checkoutNetwork = async () => {
  return new Promise((resolve) => {
    postDataToiOS(`checkoutNetwork`);
    window.rtmobile.callbacks.afterCheckoutNetwork = resolve;
  });
};
var setScreenHorizontal = async () => {
  return new Promise((resolve) => {
    postDataToiOS(`setScreenHorizontal`);
    setTimeout(() => {
      resolve();
    }, 300);
  });
};
var setScreenPortrait = async () => {
  return new Promise((resolve) => {
    postDataToiOS(`setScreenPortrait`);
    setTimeout(() => {
      resolve();
    }, 300);
  });
};
var scan = async () => {
  return new Promise((resolve) => {
    postDataToiOS(`scan`);
    window.rtmobile.callbacks.afterScan = resolve;
  });
};

// src/android.ts
var getDeviceInfo2 = async () => {
  return new Promise((resolve) => {
    window.Android.getDeviceInfo();
    window.rtmobile.callbacks.afterGetDeviceInfo = resolve;
  });
};
var writeLocal2 = async (key, value, seconds) => {
  return new Promise((resolve) => {
    window.Android.writeLocal(key, `${value}`, seconds);
    window.rtmobile.callbacks.afterWriteLocal = resolve;
  });
};
var readLocal2 = async (key) => {
  return new Promise((resolve) => {
    window.Android.readLocal(key);
    window.rtmobile.callbacks.afterReadLocal = resolve;
  });
};
var preDial2 = async (number) => {
  window.Android.preDial(number);
};
var checkoutNetwork2 = async () => {
  return new Promise((resolve) => {
    window.Android.checkoutNetwork();
    window.rtmobile.callbacks.afterCheckoutNetwork = resolve;
  });
};
var setScreenHorizontal2 = async () => {
  return new Promise((resolve) => {
    window.Android.setScreenHorizontal();
    setTimeout(() => {
      resolve();
    }, 300);
  });
};
var setScreenPortrait2 = async () => {
  return new Promise((resolve) => {
    window.Android.setScreenPortrait();
    setTimeout(() => {
      resolve();
    }, 300);
  });
};
var scan2 = async () => {
  return new Promise((resolve) => {
    window.Android.scan();
    window.rtmobile.callbacks.afterScan = resolve;
  });
};

// src/index.ts
var rtm = {
  // 初始化铺垫
  init() {
    const color1 = "#1e90ff";
    const color2 = "#fff";
    console.log(
      `%c rtlink-mbridge %c v${package_default.version} `,
      `background:${color1}; color:${color2}; padding:2px 4px; border-radius:4px 0 0 4px; border: solid 1px ${color1};`,
      `background:${color2}; color:${color1}; padding:2px 4px; border-radius:0 4px 4px 0; border: solid 1px ${color1};`
    );
    if (!window.rtmobile) {
      window.rtmobile = {
        version: package_default.version,
        callbacks: {}
      };
    }
  }
};
var index_default = rtm;
var getDeviceInfo3 = async () => {
  if (isIOS()) {
    return getDeviceInfo();
  } else if (isAndroid()) {
    return getDeviceInfo2();
  } else {
    return Promise.resolve("X");
  }
};
var writeLocal3 = async (key, value, seconds) => {
  const validityPeriod = 3600 * 24 * 365 * 99;
  if (isIOS()) {
    return writeLocal(key, value, seconds ?? validityPeriod);
  } else if (isAndroid()) {
    return writeLocal2(key, value, seconds ?? validityPeriod);
  } else {
    return Promise.resolve("X");
  }
};
var readLocal3 = async (key) => {
  if (isIOS()) {
    return readLocal(key);
  } else if (isAndroid()) {
    return readLocal2(key);
  } else {
    return Promise.resolve("X");
  }
};
var preDial3 = async (number) => {
  if (isIOS()) {
    return preDial(number);
  } else if (isAndroid()) {
    return preDial2(number);
  } else {
    return;
  }
};
var checkoutNetwork3 = async () => {
  if (isIOS()) {
    return checkoutNetwork();
  } else if (isAndroid()) {
    return checkoutNetwork2();
  } else {
    return Promise.resolve("X");
  }
};
var setScreenHorizontal3 = async () => {
  if (isIOS()) {
    return setScreenHorizontal();
  } else if (isAndroid()) {
    return setScreenHorizontal2();
  } else {
    return Promise.resolve();
  }
};
var setScreenPortrait3 = async () => {
  if (isIOS()) {
    return setScreenPortrait();
  } else if (isAndroid()) {
    return setScreenPortrait2();
  } else {
    return Promise.resolve();
  }
};
var scan3 = async () => {
  if (isIOS()) {
    return scan();
  } else if (isAndroid()) {
    return scan2();
  } else {
    return Promise.resolve("X");
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  checkoutNetwork,
  getDeviceInfo,
  isAndroid,
  isIOS,
  preDial,
  readLocal,
  scan,
  setScreenHorizontal,
  setScreenPortrait,
  writeLocal
});
